content/en/tracing/trace_collection/custom_instrumentation/ios/otel.md

```diff
@@ -0,0 +1,341 @@
+)
+```
+{{% /tab %}}
+{{< /tabs >}}
+{{< /site-region >}}
+
+{{< site-region region="us3" >}}
+{{< tabs >}}
+{{% tab "Swift" %}}
+
+```swift
+import DatadogCore
+
+Datadog.initialize(
+    with: Datadog.Configuration(
+        clientToken: "<client token>",
+        env: "<environment>",
+        site: .us3,
+        service: "<service name>"
+    ),
+    trackingConsent: trackingConsent
+)
+```
+{{% /tab %}}
+{{< /tabs >}}
+{{< /site-region >}}
+
+{{< site-region region="us5" >}}
+{{< tabs >}}
+{{% tab "Swift" %}}
+
+```swift
+import DatadogCore
+
+Datadog.initialize(
+    with: Datadog.Configuration(
+        clientToken: "<client token>",
+        env: "<environment>",
+        site: .us5,
+        service: "<service name>"
+    ),
+    trackingConsent: trackingConsent
+)
+```
+{{% /tab %}}
+{{< /tabs >}}
+{{< /site-region >}}
+
+{{< site-region region="gov" >}}
+{{< tabs >}}
+{{% tab "Swift" %}}
+
+```swift
+import DatadogCore
+
+Datadog.initialize(
+    with: Datadog.Configuration(
+        clientToken: "<client token>",
+        env: "<environment>",
+        site: .us1_fed,
+        service: "<service name>"
+    ),
+    trackingConsent: trackingConsent
+)
+```
+{{% /tab %}}
+{{< /tabs >}}
+{{< /site-region >}}
+
+{{< site-region region="ap1" >}}
+{{< tabs >}}
+{{% tab "Swift" %}}
+
+```swift
+import DatadogCore
+
+Datadog.initialize(
+    with: Datadog.Configuration(
+        clientToken: "<client token>",
+        env: "<environment>",
+        site: .ap1,
+        service: "<service name>"
+    ),
+    trackingConsent: trackingConsent
+)
+```
+{{% /tab %}}
+{{< /tabs >}}
+{{< /site-region >}}
+
+To be compliant with the GDPR regulation, the SDK requires the `trackingConsent` value at initialization.
+The `trackingConsent` can be one of the following values:
+
+- `.pending`: The SDK starts collecting and batching the data but does not send it to Datadog. The SDK waits for the new tracking consent value to decide what to do with the batched data.
+- `.granted`: The SDK starts collecting the data and sends it to Datadog.
+- `.notGranted`: The SDK does not collect any data: logs, traces, and RUM events are not sent to Datadog.
+
+To change the tracking consent value after the SDK is initialized, use the `Datadog.set(trackingConsent:)` API call.
+
+The SDK changes its behavior according to the new value. For example, if the current tracking consent is `.pending`:
+
+- If changed to `.granted`, the SDK send all current and future data to Datadog;
+- If changed to `.notGranted`, the SDK wipe all current data and stop collecting any future data.
+
+Before data is uploaded to Datadog, it is stored in cleartext in the cache directory (`Library/Caches`) of your [application sandbox][12]. The cache directory cannot be read by any other app installed on the device.
+
+When writing your application, enable development logs to log to console all internal messages in the SDK with a priority equal to or higher than the provided level.
```

Comment:
This is quite lengthy to have embedded in the instructions. I think we could pull this out into a brand new section, something like "GDPR compliance and tracking consent". Then, we can just put a note that links to the section here.

Reply:
this is common pattern we have been using at other places like https://docs.datadoghq.com/tracing/trace_collection/automatic_instrumentation/dd_libraries/ios/?tab=cocoapods

changing this would lead to further refactor. We should do it as a separate change if you think strongly.

Reply:
Sounds good. Let's leave it as is for now. 😄